WEB/wwwroot/controller/Cadastro.controller.js

```diff
@@ -66,104 +77,55 @@ sap.ui.define([
 
         setarValorPadraoInputs: function(){
             const valorPadrao = "None";
-            const  string_vazia = "";
             let campos = ["nome", "descricao", "categoria", "dataDeFabricacao", "estoque"]
             
             campos.forEach(res =>{
                 campodefinido = this.getView().byId(res)
                 campodefinido.setValueState(valorPadrao)
-                campodefinido.setValue(string_vazia)
+                campodefinido.setValue(stringVazia)
             })
         },
 
-        aoClicarSalvar: function () {
-            const peca = this.getView()
-                .getModel(modeloPeca)
-                .getData();
-                
-            this.validarCampos(peca);
-            
-            const campoData = this.getView().byId(idDataFabricacao);
-            
-            if(Validacao.ehCamposValidos(peca, campoData)){
-                peca.id
-                    ?this._editarPeca(peca)
-                    :this._salvarPeca(peca);
-            }
-        }, 
-
         validarCampos: function(peca){
-            const propId = "id";
+            ProcessaEvento.processarEvento(() => {
```

Comment:
não era bem isso, fica redundante se você colocar dentro de um serviço, o nome se repete. O que seria interessante é colocar dentro de uma classe base que extende pras suas controllers, isso vai permitir você chamar o processarEvento usando um this. - Tenda fazer, se não der certo você me chama